docs/whats_new.rst

```diff
@@ -5,255 +5,290 @@ What's New
 ##########
 
 **************************
-What's New in FLARE v2.3.0
+What's New in FLARE v2.4.0
 **************************
```

Comment:
Instead of every release:
1. Move the content of whats_new.rst to a flare_[old version] release note file
2. Re-write this "whats_new.rst"

I suggest we instead do:
1. Create a release note rst file flare_[version].rst
2. Change whats_new.rst to include that new version release note file

This way we are NOT modifying a lot of contents in this entry file "whats_new.rst"
It will be easier to track the changes in GitHub

Reply:
Good suggestion, I'll change to this